File: HomeWork/期末專題作業/C110152338/Frontend/src/app/course.ts
export interface Course {
  courseId: number;
  courseName: string;
  description: string;
  instructor: string;
  price: number;
  startDate: string; // 修正型別為 string
  classroom: string; // 新增 classroom 屬性
  schedule: string; // 新增 schedule 屬性
}

File: HomeWork/期末專題作業/C110152338/Backend/Models/Course.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Course
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CourseId { get; set; }

        [Required]
        public required string CourseName { get; set; }

        public required string Description { get; set; }

        public required string Instructor { get; set; }

        public string Classroom { get; set; } = string.Empty;
        public string Schedule { get; set; } = string.Empty;

        [Column(TypeName = "decimal(10, 2)")]
        public decimal Price { get; set; }

        public DateTime? StartDate { get; set; }
    }
}